File: 2024/dec01/02.py
ids = open('input.txt', 'r')
id1 = []
id2 = []
for line in ids:
    x = line.split('   ')
    id1.append(int(x[0][:5]))
    id2.append(int(x[1][:5]))

appearance, timesFound = 0, 0
for i1 in id1:
    for i2 in id2:
        if i1==i2:
            timesFound+=1
    appearance+= i1*timesFound
    timesFound=0

print('Similarity score (challenge 2):', appearance)
#Right answer: 23046913

File: 2024/dec01/C++/02.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    vector<int> id1 {};
    vector<int> id2 {};
    int sim {0};

    ifstream file("../input.txt");
    if (!file) return 1;

    int n1, n2;
    while (file >> n1 >> n2) {
        id1.push_back(n1);
        id2.push_back(n2);
    }

    for (int n : id1) {
        int c = count(id2.begin(), id2.end(), n);
        sim += n*c;
    }

    cout << "Similarity score (challenge 2): " << sim << endl;
    //Right answer: 23046913

    return 0;
}